Crawler/SiteScraper/ScraperDefinition.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Crawler.SiteScraper
+{
+    public class ScraperDefinition
+    {
+        private Regex regex = new Regex(string.Empty);
+
+        public string Selector { get; set; }
+
+        public string Pattern
+        {
+            get => regex.ToString();
+            set => regex = new Regex(value);
+        }
+
+        public string? Match(string url)
+        {
+            var match = regex.Match(url);
+
+            return match.Success ? Selector : null;
+        }
+    }
```

Comment:
This class is very similar to "TokenDefinition". I would consider extract a base class ("RegexDefinition").

Reply:
Even tough they similar, they serve completely different use case and logic.
The only thing in common is that they have pattern and index, even the Match method has different implementation for them.
I don't think that they should have a base class because they do not depend to used together, they just have regex in them that's it.

Reply:
Alright, accepted.